### PR TITLE
ref(django-3.2): Vendor django picklefield

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Django-picklefield hasn't been updated in 2 years, but we need it. We also need to upgrade to django 3.2 which means we need to update picklefield.

Could use some direction on the direction struture changes

### CODE REVIEW

**Comment by AniketDas-Tekky** [comment]
> Where are we using this? How feasible would it be to kill off usage of pickle in favor of JSON?

We use it in authentication and all option models. I think we should remove this, but it's outside the scope of django 3.2 upgrade.

File: src/django-picklefield/constants.py [hunk 1]
```diff
@@ -0,0 +1 @@
+DEFAULT_PROTOCOL = 2
```
Comment (asottile-sentry) [inline src/django-picklefield/constants.py:1]: the package name needs to be an underscore -- that's why the testsuite is failing

**Comment by chadwhitacre** [comment]
Can we fork and maintain?

**Comment by AniketDas-Tekky** [comment]
> Can we fork and maintain?

Long term, this is the right approach (or removing it entirely). However, for django 3.2 upgrade, we only need to update a couple of lines to make it compatible. So this is a quick fix, until we get some time to set up our own fork.

FWIW, I did fork it already, but there's no CI. https://github.com/getsentry/django-picklefield

**Comment by chadwhitacre** [comment]
> we only need to update a couple of lines to make it compatible

Can we make the changes in the fork and reference it via a git URL in requirements.txt? Avoids the overhead of releasing but keeps us on the right path w/ the fork.

**Comment by chadwhitacre** [comment]
Also not _that_ hard to set up releases via [the `publish` repo](https://github.com/getsentry/publish). ;)

**Comment by asottile-sentry** [comment]
> > we only need to update a couple of lines to make it compatible
> 
> Can we make the changes in the fork and reference it via a git URL in requirements.txt? Avoids the overhead of releasing but keeps us on the right path w/ the fork.

at the moment our requirements require `pkg==...` and forbid git urls (because we can't publish with them)

we could publish a `sentry-django-picklefield` though

**Comment by joshuarli** [comment]
I still think vendoring is the easiest and best option here, given that it's such a small amount of code and is easy to vendor. It's also MIT license, no issues there I don't think.

Granted, we should try to also upstream any improvements and compat changes we make.

**Review by asottile-sentry — APPROVED** [review]

File: src/django_picklefield/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import django.utils.version
```
Comment (joshuarli) [inline src/django_picklefield/__init__.py:1]: You can remove the version related contents of this file, it's not needed anymore.

**Review by joshuarli — APPROVED** [review]

**Pull request merged**